src/base/language_support.c

```diff
@@ -3789,7 +3789,7 @@ lang_strmatch_utf8_uca_w_level (const COLL_DATA * coll_data, const int level, bo
 	  do
 	    {
 	      w2 = GET_UCA_WEIGHT (level, ce_index2, uca_w_l13_2, uca_w_l4_2);
-	      if (w2 != 0)
+	      if (w2 != 0 || !ignore_trailing_space)
```

Comment:
Is it correct to result = -1 regardless of the value of w2?
If so, I don't think the condition needs to be here.

Reply:
because the string1's length is already less than string2's

Reply:
actually the code is prior to the "do... " as like below
```
if (!ignore_trailing_space)
  {
     return -1;
  }
```
I move the code to simplify the code line.